industry_benchmarks/utils/data_gathering.py

```diff
@@ -0,0 +1,982 @@
+    rmsd_scorer = MappingRMSDScorer()
+    score = rmsd_scorer(mapping=mapping)
+    return score
+
+
+def get_number_heavy_dummy_heavy_core_atoms(mapping: LigandAtomMapping) -> int:
+    """
+    This function calculates the size of the mapped and unmapped regions of
+    an edge.
+
+    Returns
+    -------
+    size_core: int
+      The number of heavy atoms in the mapped (core) region of the hybrid topology.
+    size_dummy_A: int
+      The number of heavy atoms in the unmapped (dummy) region of compound A
+    size_dummy_B: int
+      The number of heavy atoms in the unmapped (dummy) region of compound B
+    """
+    molA = mapping.componentA.to_rdkit()
+    molB = mapping.componentB.to_rdkit()
+    mapped_atomIDs = mapping.componentA_to_componentB
+
+    heavy_core_A = []
+    heavy_dummy_A = []
+    heavy_core_B = []
+    heavy_dummy_B = []
+
+    for atom in molA.GetAtoms():
+        # For heavy core: Heavy atom indices that are in the mapping:
+        if atom.GetIdx() in mapped_atomIDs.keys() and atom.GetAtomicNum() != 1:
+            # Check if the atom is mapped to a hydrogen atom
+            index_mapping = list(mapped_atomIDs.keys()).index(atom.GetIdx())
+            index_atomB = list(mapped_atomIDs.values())[index_mapping]
+            atomic_number_mapped_atom_B = molB.GetAtoms()[index_atomB].GetAtomicNum()
+            # If the mapped atom in ligand B is a hydrogen, these atoms are unmapped.
+            # Therefore we will could the heavy atom as a dummy atom, not core
+            if atomic_number_mapped_atom_B == 1:
+                heavy_dummy_A.append(atom.GetIdx())
+            else:
+                heavy_core_A.append(atom.GetIdx())
+        # For heavy dummy: Heavy atom indices that are not in the mapping:
+        if atom.GetIdx() not in mapped_atomIDs.keys() and atom.GetAtomicNum() != 1:
+            heavy_dummy_A.append(atom.GetIdx())
+
+    for atom in molB.GetAtoms():
+        # For heavy core: Heavy atom indices that are in the mapping:
+        if atom.GetIdx() in mapped_atomIDs.values() and atom.GetAtomicNum() != 1:
+            # Check if the atom is mapped to a hydrogen atom
+            index_mapping = list(mapped_atomIDs.values()).index(atom.GetIdx())
+            index_atomA = list(mapped_atomIDs.keys())[index_mapping]
+            atomic_number_mapped_atom_A = molA.GetAtoms()[index_atomA].GetAtomicNum()
+            # If the mapped atom in ligand B is a hydrogen, these atoms are unmapped.
+            # Therefore we will could the heavy atom as a dummy atom, not core
+            if atomic_number_mapped_atom_A == 1:
+                heavy_dummy_B.append(atom.GetIdx())
+            else:
+                heavy_core_B.append(atom.GetIdx())
+        # For heavy dummy: Heavy atom indices that are not in the mapping:
+        if atom.GetIdx() not in mapped_atomIDs.values() and atom.GetAtomicNum() != 1:
+            heavy_dummy_B.append(atom.GetIdx())
+
+    assert len(heavy_core_A) == len(heavy_core_B)
```

Comment:
Just a comment, not something we need to do anything about: 

It's good to have that check here, and generally we shouldn't see this fail - however we should keep in mind that the user facing error here could be confusing.